Docs/checklist-for-new-user-facing-features.md

```diff
@@ -0,0 +1,47 @@
+## New Feature Development Checklist
+
+So you've added a new feature. Users will see it!
+
+You've landed the code, it's indented properly, it looks lovely! What's Next!?
+
+###  Has your feature got tests yet?
+
+  * perhaps existing tests will cover it, may be adding another unit test will be enough.
+  * If you're adding new UI:
+    - ensure widgets have `accessbilityIdentifiers`. 
+    - Add it to FxScreenGraph.swift.
+    - XCUITests should be used, but as a last resort.
+
+### Should your new feature be keyboard navigable?
+
+  * Adding UIKeyCommands will help.
+  * Getting to the new features is as important as using your feature.
+
+### Accessibility for new UI
+
+  * Add accessibilityLabels
+  * Have you tested it with VoiceOver?
+
+### Has your feature got new Strings?
+
+  * Is it `NSLocalizable`? 
+  * It should probably go in `Strings.swift`. Ask about String export deadlines.
+  * Are the strings screenshotted? The `L10nScreenshotTests` are there to help diagnose layout issues caused by missing or long strings and right to left issues.
+
+### How do you know if your new feature is used?
+
+  * Add event telemetry
+  * Consider performance telemetry.
+  * Add a Telemetry dashboard — this will help make decisions about your feature in the future.
```

Comment:
I think that when you include telemetry for a new feature, we should also try to make it a habit to immediately create a dashboard or at least a very simple query. I think we should have a dashboard that simply has one query per feature or telemetry attribute set so that you (and QA) can very easily verify that the telemetry is working as expected.

Reply:
The need for this has become more clear after we had to do an emergency fix for the screen size telemetry. We would have caught that sooner had we added a simple query to that dashboard.

I'm not thinking 'deep insight' here - just a query that returns a table with the new attributes that we collect.